packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts

```diff
@@ -0,0 +1,425 @@
+
+        const result = await clipPolygon(circles);
+
+        // Create a MultiPolygon GeoJSON and verify it's valid in PostGIS
+        const multiPolygon = {
+            type: 'MultiPolygon',
+            coordinates: result
+        };
+
+        const validation = await knex.raw(`
+            SELECT
+                ST_IsValid(ST_GeomFromGeoJSON(?)) as is_valid,
+                ST_GeometryType(ST_GeomFromGeoJSON(?)) as geom_type,
+                ST_Area(ST_GeomFromGeoJSON(?)::geography) as area_sqm
+        `, [JSON.stringify(multiPolygon), JSON.stringify(multiPolygon), JSON.stringify(multiPolygon)]);
+
+        expect(validation.rows[0].is_valid).toBe(true);
+        expect(validation.rows[0].geom_type).toBe('ST_MultiPolygon');
+
+        const areaSqM = parseFloat(validation.rows[0].area_sqm);
+        // Area should be approximately π * 500²
+        const expectedArea = Math.PI * 500 * 500;
+        expect(areaSqM).toBeGreaterThan(expectedArea * 0.95);
+        expect(areaSqM).toBeLessThan(expectedArea * 1.05);
+    });
+
+    test('should handle concurrent calls without conflicts', async () => {
+        const circles1 = [
+            { center: [-73.5, 45.5] as [number, number], radiusKm: 0.5 }
+        ];
+        const circles2 = [
+            { center: [-73.6, 45.6] as [number, number], radiusKm: 0.3 }
+        ];
+        const circles3 = [
+            { center: [-73.7, 45.7] as [number, number], radiusKm: 0.4 }
+        ];
+
+        // Call multiple times in parallel
+        const results = await Promise.all([
+            clipPolygon(circles1),
+            clipPolygon(circles2),
+            clipPolygon(circles3)
+        ]);
+
+        // All should succeed
+        expect(results.length).toBe(3);
+        results.forEach((result) => {
+            expect(Array.isArray(result)).toBe(true);
+            expect(result.length).toBe(1);
+        });
+    });
+
+    test('should handle transaction properly and clean up temp tables', async () => {
+        const circles = [
+            { center: [-73.5, 45.5] as [number, number], radiusKm: 0.5 }
+        ];
+
+        // Execute multiple times
+        await clipPolygon(circles);
+        await clipPolygon(circles);
+        await clipPolygon(circles);
+
+        // Check that no temp tables are left behind
+        // Note: temp tables auto-drop on commit, so there should be none or very few
+        const tempTables = await knex.raw(`
+            SELECT COUNT(*) as count
+            FROM pg_tables
+            WHERE schemaname LIKE 'pg_temp%'
+            AND tablename LIKE 'temp_circles_%'
+        `);
+
+        // Should be 0 or very low if other tests are running in parallel
+        expect(parseInt(tempTables.rows[0].count)).toBeLessThan(10);
+    });
+
+    test('should throw error when database connection fails', async () => {
+        // This test ensures error handling works
+        // We can't easily simulate DB failure, but we can test the error path exists
```

Comment:
This does not seem to test error failure at all. You can probably send an invalid value as parameter, with a cast as any and make sure it throws an error (if it's the case)